src/views/sandwich-view.tsx

```diff
@@ -18,7 +18,53 @@ interface SandwichViewProps {
   glCanvas: HTMLCanvasElement
 }
 
-class SandwichView extends StatelessComponent<SandwichViewProps> {
+interface SandwichViewState {
+  isDragging: boolean
+  resizerPos: number | null
+  initialPos: number
+  delta: number
+}
+
+class SandwichView extends Component<SandwichViewProps, SandwichViewState> {
+  constructor(props: SandwichViewProps) {
+    super(props)
+    this.state = {
+      isDragging: false,
+      resizerPos: null,
+      initialPos: 0,
+      delta: 0,
+    }
+  }
+
+  startResize = (e: MouseEvent) => {
+    this.setState({
+      isDragging: true,
+      initialPos: e.clientX,
+      delta: 0,
+    })
+  }
+
+  resizeViews = (e: MouseEvent) => {
+    if (this.state.isDragging) {
+      let delta = e.clientX - this.state.initialPos
+      this.setState({delta: delta})
+    }
+  }
+
+  stopResize = (e: MouseEvent) => {
+    if (this.state.isDragging) {
+      let resizerPos = this.state.resizerPos
+      if (!resizerPos) {
+        resizerPos = this.state.initialPos
+      }
+      this.setState({
+        isDragging: false,
+        resizerPos: resizerPos + this.state.delta,
+        delta: 0,
+      })
```

Comment:
As far as I can tell from this code, the resizing only happens on mouse up. I think it's a much better user experience to have continuous feedback during dragging.